src/pay/components/PayProvider.tsx

```diff
@@ -0,0 +1,281 @@
+      },
+      enabled: !!transactionId,
+    },
+  });
+  const transactionHash = data?.receipts?.[0]?.transactionHash;
+  const { data: receipt } = useWaitForTransactionReceipt({
+    hash: transactionHash,
+  });
+
+  // Component lifecycle emitters
+  useEffect(() => {
+    onStatus?.(lifecycleStatus);
+  }, [
+    lifecycleStatus,
+    lifecycleStatus.statusData, // Keep statusData, so that the effect runs when it changes
+    lifecycleStatus.statusName, // Keep statusName, so that the effect runs when it changes
+    onStatus,
+  ]);
+
+  // Set transaction pending status when writeContracts is pending
+  useEffect(() => {
+    if (status === 'pending') {
+      updateLifecycleStatus({
+        statusName: PAY_LIFECYCLESTATUS.PENDING,
+        statusData: {},
+      });
+    }
+  }, [status, updateLifecycleStatus]);
+
+  // Trigger success status when receipt is generated by useWaitForTransactionReceipt
+  useEffect(() => {
+    if (!receipt) {
+      return;
+    }
+    updateLifecycleStatus({
+      statusName: PAY_LIFECYCLESTATUS.SUCCESS,
+      statusData: {
+        transactionReceipts: [receipt],
+        chargeId: chargeId,
+        receiptUrl: `https://commerce.coinbase.com/pay/${chargeId}/receipt`,
+      },
+    });
+  }, [chargeId, receipt, updateLifecycleStatus]);
+
+  // biome-ignore lint/complexity/noExcessiveCognitiveComplexity: TODO Refactor this component to deprecate funding flow
+  const handleSubmit = useCallback(async () => {
+    try {
+      // Open Coinbase Commerce receipt
+      if (lifecycleStatus.statusName === PAY_LIFECYCLESTATUS.SUCCESS) {
+        window.open(
+          `https://commerce.coinbase.com/pay/${chargeId}/receipt`,
+          '_blank',
+          'noopener,noreferrer',
+        );
+        return;
+      }
+      // Open funding flow
+      // TODO: Deprecate this once we have USDC Magic Spend
+      if (
+        lifecycleStatus.statusName === PAY_LIFECYCLESTATUS.ERROR &&
+        lifecycleStatus.statusData?.code === PayErrorCode.INSUFFICIENT_BALANCE
+      ) {
+        window.open(
+          'https://keys.coinbase.com/fund',
+          '_blank',
+          'noopener,noreferrer',
+        );
+        return;
+      }
+
+      let connectedAddress = address;
+      let connectedChainId = chainId;
+      if (!isConnected) {
+        // Prompt for wallet connection
+        // This is defaulted to Coinbase Smart Wallet
+        const { accounts, chainId: _connectedChainId } = await connectAsync({
+          connector:
+            connectors.find(
+              (connector) => connector.id === 'coinbaseWalletSDK',
+            ) || coinbaseWallet({ preference: 'smartWalletOnly' }),
+        });
+        connectedAddress = accounts[0];
+        connectedChainId = _connectedChainId;
+      }
+
+      // This shouldn't ever happen, but added this to make Typescript happy
+      /* v8 ignore next 3 */
+      if (!connectedAddress) {
+        return;
```

Comment:
I think maybe we can add an `updateLifecycleStatus` error message here, but almost always there will be `connectedAddress` returned from `connectAsync`